src/lib.rs

```diff
@@ -537,6 +540,78 @@ where
 
         Err(GptError::NotEnoughSpace)
     }
+    /// create a new partition with a specific id
+    /// a specific name
+    /// a specific first_lba
+    /// a specific length_lba
+    /// a specific part_type
+    /// a specific flags
+    /// ## Panics
+    /// If length is empty panics
+    /// If id zero panics
+    pub fn new_partition(
+        &mut self,
+        name: &str,
+        id: u32,
+        first_lba: u64,
+        length_lba: u64,
+        part_type: partition_types::Type,
+        flags: u64,
+    ) -> Result<u32, GptError> {
+        assert!(length_lba > 0, "length must be greater than zero");
+        assert!(id > 0, "id must be greater than zero");
+        //check id
+        match self.partitions.get(&id) {
+            Some(p) if p.is_used() => return Err(GptError::PartitionIdAlreadyUsed),
+            /// Allow unused ids , because we can allow to modify the part count
+            None => {}
+            _ => {
+                // TODO I don't know what will happen in this scope
```

Comment:
I think you can just combine None and Some(_) with `_`. Since add partition already overrides unused partitions.

Reply:
Im only check for possible override on existed partitions here and do other check below since the given id can be a truly unknown one.I must do full check for the given id.I just dont know what happened if _ return.

Reply:
Im only check for possible override on existed partitions here and do other checks below since the given id can be a truly unknown one.I must do full checks for the given id.

Reply:
I think I wasn't clear. I get why the match statement exists. I'm just talking about your todo, You can remove the None pattern, and add a comment if wanted stating, every partition which is not used might be used. That already happens in `add_partition` so this should not be a behavioural change.

Reply:
Thanks for your guidance.Now with test and fmt,I have enlarged the mem-disk size in your test code to hold the test for add_partition_at()